### DISCUSSION TITLE
지하철 구간 추가 시 순서에 대한 질문

### DISCUSSION THREAD

**Discussion by hanse-kim** [woowacourse/java-subway-map-precourse#8] [Q&A] [answered]
"순서는 1부터 시작한다."라는 설명의 의미가 궁금해서 질문드립니다.
만약 2호선에 교대역 - 강남역 - 역삼역이 존재하고 2호선의 순서 2에 잠실역을 추가한다고 할 때,

- **case 1**
```
 (1)      (2)      (3)
교대역 - 강남역 - 역삼역

결과: 교대역 - 잠실역 - 강남역 - 역삼역
```
- **case 2**
```
      (1)      (2)
교대역 - 강남역 - 역삼역

결과: 교대역 - 강남역 - 잠실역 - 역삼역
```

어느 쪽이 출제 의도에 맞을까요?

**Comment by Livenow14** [comment]
상행, 하행 종점역도 바뀔 수 있다는 부분애서
저는 이렇게 생각했어요!
```
(1)      (2)      (3)     (4)
 - 교대역 - 강남역 - 역삼역 -

```
순서 2에 잠실을 추가한다면 case 1번이지 않을까요?!

**Reply by hanse-kim** [reply]
구간 추가에서는 "역과 역 사이에 새로운 역이 추가 될 수 있다."라고 되어 있어서 저는 종점역을 새로 추가할 수는 없다고 생각했어요! 종점은 역과 역 사이가 아니니..
그거랑 별개로 case 1은 이렇게 표현하는게 더 적절하네요. 저는 여기서 말하는 순서가 역의 순서인지 구간의 순서인지 헷갈렸는데, 생각해보면 구간의 순서로 보는 게 당연하고 질문처럼 표기하면 (4)를 둘 위치가 애매해지네요.
답변 감사합니다 :)

**Reply by pkeugine** [reply]
안녕하세요! 저도 이 부분이 조금 헷갈렸는데 제가 이해한 방식을 공유해볼까 합니다!

저도 @Livenow14 분과 동일하게 생각했는데, 그 이유는 :

1. 구간 삭제 기능에서 종점역을 삭제할 수 있다.
2. 종점을 추가해도 새로운 구간이 생긴다:

(새로운 종점)--새로운 구간--(이전 종점)---(강남역)

3. "역과 역 사이에 새로운 역이 추가될 수 있다"를 "역과 역 사이에도 새로운 역이 추가될 수 있다"로 이해했다. 즉, 역은 꼭 노선의 끝 부분에만 추가할 수 있는게 아니다.

물론 이게 맞지 않을 수도 있지만 도움이 되었으면 좋겠네요 :) 다른 분들도 의견 있으시면 피드백 환영합니다!

**Reply by hanse-kim** [reply]
@pkeugine 아하.. 참고가 되었습니다! 감사합니다 :)

**Comment by gracefulBrown** [comment, marked as answer]
`순서는 1부터 시작한다.`는 순서로 입력할 수 있는 숫자가 0이 아닌 1부터 가능하다는 의미 입니다.

**Reply by hanse-kim** [reply]
확인했습니다. 답변 감사합니다 :)
